testing/bdd/test_ebuttd_style_references.py

```diff
@@ -0,0 +1,67 @@
+from pytest_bdd import scenarios, when, then, given, parsers
+import xml.etree.ElementTree as ET
+
+scenarios('features/styles/ebuttd_style_references.feature')
+
+
+@when(parsers.parse('it contains style "{style_name}"'))
+def when_it_contains_style(test_context, template_dict, style_name):
+    if 'styles' not in template_dict:
+        template_dict['styles'] = list()
+    style = {"id": style_name}    
+    template_dict['styles'].append(style)
+    test_context[style_name] = style
+
+
+@when(parsers.parse('style "{style_name}" has attribute "{attribute}" set to "{value}"'))
+def when_style_has_attribute(test_context, style_name, attribute, value):
+    test_context[style_name][attribute] = value
+
+
+@when(parsers.parse('it contains some text with style "{style_name}"'))
+def when_style_has_attribute(template_dict, style_name):
```

Comment:
This and some of the other function names don't match what the step does - please could you change them, e.g. in this case `when_text_has_style` could work.

Reply:
Good spot!